SmartSaver.Domain/Repositories/Transaction/TransactionRepo.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using SmartSaver.EntityFrameworkCore.Models;
+using SmartSaver.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+using System.Threading.Tasks;
+using SmartSaver.Domain.CustomExceptions;
+
+namespace SmartSaver.Domain.Repositories
+{
+    public class TransactionRepo : ITransactionRepo
+    {
+        public TransactionRepo(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        private readonly ApplicationDbContext _context;
+
+        public List<TransactionDto> GetThisMonthAccountSpendings(int accId)
+        {
+            var transactions = _context.Transactions
+                .Include(p => p.Category)
+                .Where(t => t.ActionTime > CurrentMonthFirstDayDate() &&
+                            t.AccountId == accId
+                            && t.Category.TypeOfIncome == false).ToList();
+            return transactions;
+        }
+
+        public List<Balance> GetBalanceHistory(int accId)
+        {
+            var chartData = new List<Balance>();
+
+            var transactions = _context.Transactions.Include(nameof(CategoryDto))
+                .Where(t => t.ActionTime > CurrentMonthFirstDayDate() && t.AccountId == accId);
+
+            decimal balance = 0;
+            foreach (var t in transactions)
+            {
+                balance += t.Amount;
+                chartData.Add(new Balance()
+                {
+                    Amount = balance,
+                    ActionTime = t.ActionTime
+                });
+            }
+
+            return chartData;
+        }
```

Comment:
Just group account transactions by DateTime. A dictionary can be returned here with action time and amount. Also this method can be moved to TransactionCounterService where account transactions can be passed as the argument